runtime-modules/storage/src/data_directory.rs

```diff
@@ -369,14 +443,23 @@ decl_module! {
             // Ensure global voucher constraints satisfied.
             Self::ensure_global_voucher_constraints_satisfied(upload_voucher_delta)?;
 
+            // fill vouchers - overflow cannot happen because of prior constraint checks
+            let new_owner_voucher = owner_voucher.fill_voucher::<T>(upload_voucher_delta)?;
+            let new_global_voucher = Self::global_voucher().fill_voucher::<T>(upload_voucher_delta)?;
```

Comment:
This checks have been already performed in `ensure_global_voucher_constraints_satisfied` function.
I understand the purpose of having separate functions for both size and objects limit, though i guess it would be better to separate mutation from checks here and reuse the separate checks inside the `ensure_global_voucher_constraints_satisfied`.

Reply:
Yes I see the redundant check here which I did highlight with the comment. I'll try to refactor.

Reply:
Fixed in 8126da5